src/coreComponents/physicsSolvers/multiphysics/PoromechanicsSolver.hpp

```diff
@@ -0,0 +1,270 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PoromechanicsSolver.hpp
+ *
+ */
+
+#ifndef GEOS_PHYSICSSOLVERS_MULTIPHYSICS_POROMECHANICSSOLVER_HPP_
+#define GEOS_PHYSICSSOLVERS_MULTIPHYSICS_POROMECHANICSSOLVER_HPP_
+
+#include "physicsSolvers/multiphysics/CoupledSolver.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.hpp"
+#include "constitutive/solid/PorousSolid.hpp"
+//#include "dataRepository/Group.hpp"
+
+namespace geos
+{
+
+template< typename FLOW_SOLVER >
+class PoromechanicsSolver : public CoupledSolver< FLOW_SOLVER,
+                                                  SolidMechanicsLagrangianFEM >
+{
+public:
+
+  using Base = CoupledSolver< FLOW_SOLVER,
+                              SolidMechanicsLagrangianFEM >;
+  using Base::m_solvers;
+  using Base::m_dofManager;
+  using Base::m_localMatrix;
+  using Base::m_rhs;
+  using Base::m_solution;
+
+  enum class SolverType : integer
+  {
+    Flow = 0,
+    SolidMechanics = 1
+  };
+
+  /// String used to form the solverName used to register solvers in CoupledSolver
+  static string coupledSolverAttributePrefix() { return "poromechanics"; }
+
+  /**
+   * @brief main constructor for CoupledSolver Objects
+   * @param name the name of this instantiation of CoupledSolver in the repository
+   * @param parent the parent group of this instantiation of CoupledSolver
+   */
+  PoromechanicsSolver( const string & name,
+                       dataRepository::Group * const parent )
+    : Base( name, parent )
+  {
+    this->registerWrapper( viewKeyStruct::useNAString(), &m_useNA ).
+      setApplyDefaultValue( 0 ).
+      setInputFlag( dataRepository::InputFlags::OPTIONAL ).
+      setDescription( "Flag to indicate that the solver is going to use nonlinear acceleration" );
+  }
+
+  struct viewKeyStruct : SolverBase::viewKeyStruct
+  {
+    /// Flag to indicate that the solver is going to use nonlinear acceleration
+    constexpr static char const * useNAString() { return "useNA"; }
+  };
+
+protected:
+
+  /* Implementation of Nonlinear Acceleration (Aitken) of averageMeanTotalStressIncrement */
+
+  void recordAverageMeanTotalStressIncrement( DomainPartition & domain,
+                                              array1d< real64 > & s )
+  {
+    // s denotes averageMeanTotalStressIncrement
+    s.resize( 0 );
```

Comment:
Maybe use a more meaningful name here instead of `s` to help readability. You could just have `savedMeanTotalStressIncrement` or `subRegionMeanTotalStressIncrement`? You could then remove the comment made above saying that `s` is the total stress increment.

Reply:
done